PWGCF/JCorran/Tasks/jflucAnalysisTask.cxx

```diff
@@ -0,0 +1,131 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+/// \author Jasper Parkkila (jparkkil@cern.ch)
+/// \author Dong Jo Kim (djkim@jyu.fi)
+/// \since Sep 2022
+
+#include "Framework/runDataProcessing.h"
+#include "Framework/AnalysisTask.h"
+#include "Framework/ASoAHelpers.h"
+#include <CCDB/BasicCCDBManager.h>
+#include "Framework/RunningWorkflowInfo.h"
+#include "Framework/HistogramRegistry.h"
+
+#include "Common/DataModel/EventSelection.h"
+#include "Common/Core/TrackSelection.h"
+#include "Common/DataModel/TrackSelectionTables.h"
+#include "Common/DataModel/Centrality.h"
+#include "ReconstructionDataFormats/V0.h"
+
+#include "PWGCF/JCorran/DataModel/JCatalyst.h"
+#include "JFFlucAnalysis.h"
+
+using namespace o2;
+using namespace o2::framework;
+using namespace o2::framework::expressions;
+
+#define O2_DEFINE_CONFIGURABLE(NAME, TYPE, DEFAULT, HELP) Configurable<TYPE> NAME{#NAME, DEFAULT, HELP};
+
+class TrackIter : public JTrackIterInterface
+{
+ public:
+  TrackIter(aod::ParticleTrack::iterator _m1) : m1(_m1){};
+  ~TrackIter(){};
+  aod::ParticleTrack::iterator m1;
+  PtEtaPhiEVector v;
+  PtEtaPhiEVector& deref()
+  {
+    v.SetPt((*m1).pt());
```

Comment:
Is this not copying the actual values, @jaelpark ?

Reply:
@jgrosseo Yeah it puts them into a ```PtEtaPhiEVector```, but at least not for the entire list at once, that I understood was not allowed from Victor's comments. But if ```PtEtaPhiEVector``` (and such) are forbidden, then I would do some heavier changes @dongjokim.

Reply:
Ideally everything is zero copy which can be achieved with some templated functions or classes but of course one cannot replay a full framework which copies things. I propose that we briefly discuss this tomorrow when we meet anyway before you go ahead and do the changes. How does that sound?

Reply:
Yep, okay thanks. Anyways I'll probably get back to this later, have to prepare for PF.